diff --git a/node_modules/nodent-transform/arboriculture.js b/node_modules/nodent-transform/arboriculture.js
index 3630ca7..a546108 100644
--- a/node_modules/nodent-transform/arboriculture.js
+++ b/node_modules/nodent-transform/arboriculture.js
@@ -786,6 +786,17 @@ function asynchronize(pr, opts, logger, parsePart, printNode) {
             } else {
                 ref.parent[ref.field] = synthBlock;
             }
+
+            const isFirst = ({type, $continuation}) => $continuation && type === 'FunctionDeclaration';
+            const isSecond = (v) => !isFirst(v);
+
+            synthBlock.body = synthBlock
+              .body
+              .filter(isFirst)
+              .concat(synthBlock
+                .body
+                .filter(isSecond)
+              )
         }
     }
     
@@ -1473,11 +1484,11 @@ function asynchronize(pr, opts, logger, parsePart, printNode) {
                     "body": body
                 }
             });
+            mapped.push(continuation);
             mapped.push({
                 type: 'ReturnStatement',
                 argument: invokeIterate
             });
-            mapped.push(continuation);
             path[0].replace(mapped.map(down));
         }
     }
