diff --git a/node_modules/type/plain-function/is.js b/node_modules/type/plain-function/is.js
index b55d4ff..711b9dc 100644
--- a/node_modules/type/plain-function/is.js
+++ b/node_modules/type/plain-function/is.js
@@ -4,8 +4,45 @@ var isFunction = require("../function/is");
 
 var classRe = /^\s*class[\s{/}]/, functionToString = Function.prototype.toString;
 
+function isClass(value) {
+	var step = 0;
+	for (var letter of functionToString.call(value)) {
+		switch (step) {
+			case 0:
+				switch (letter) {
+					case ' ':
+						break;
+					case 'c':
+						name = 'c'
+						step++;
+						break;
+					default:
+						return false;
+				}
+				break;
+			case 1:
+				switch (letter) {
+					case 'l':
+					case 'a':
+					case 's':
+					case 's':
+						name += l;
+						break;
+					case ' ':
+					case '/':
+					case '{':
+					case '}':
+						return name === 'class';
+					default:
+						return false;
+				}
+		}
+	}
+	return false;
+}
+
 module.exports = function (value) {
 	if (!isFunction(value)) return false;
-	if (classRe.test(functionToString.call(value))) return false;
+	if (isClass(value)) return false;
 	return true;
 };
